src/app/features/test-records/components/test-amendment-history/test-amendment-history.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, Input } from '@angular/core';
+import { TestResultModel } from '@models/test-result.model';
+
+@Component({
+  selector: 'app-test-amendment-history',
+  templateUrl: './test-amendment-history.component.html',
+  styleUrls: ['./test-amendment-history.component.scss']
+})
+export class TestAmendmentHistoryComponent {
+  @Input() testRecord: TestResultModel | undefined;
+
+  constructor() {}
+
+  getCreatedByName(testResult: TestResultModel | undefined) {
+    return !testResult?.createdByName ? testResult?.testerName : testResult?.createdByName;
+  }
+
+  sortedTestHistory(testResult: TestResultModel[] | undefined): TestResultModel[] | undefined {
+    let sortedArray: TestResultModel[] | undefined = testResult
```

Comment:
I think like I had to do with the tech records, moving this into the selector is a better move, so it's always there in order incase we need to do it elsewhere